Refactor rules structure

- Move rules out of `index.js` and put them all in separate files, in `index.js` keep just imports
- apply directory structure like in this example http://eslint.org/docs/developer-guide/working-with-plugins#default-configuration-for-plugins

@gvidon проверил, работает:

```
>>> eslint realhance/static/realhance/js/routes.js 

realhance/static/realhance/js/routes.js
    1:1   error    Definition for rule 'no-shadows' was not found                               no-shadows
  281:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  304:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  345:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  352:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  359:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  365:0   warning  Multiline node should have trailing new line                                 ottofeller/newline-around-multiline-blocks
  380:57  error    Block must not be padded by blank lines                                      padded-blocks
  385:5   error    Block must not be padded by blank lines                                      padded-blocks
  407:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  409:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  436:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  443:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  445:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  458:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  466:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  472:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  480:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  486:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  493:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  511:0   warning  There should not be new line between singleline nodes                        ottofeller/newline-around-multiline-blocks
  516:5   warning  Variables within the same declaration block should be sorted alphabetically  sort-vars

✖ 22 problems (3 errors, 19 warnings)
```

--- a/index.js
+++ b/index.js
@@ -5,83 +5,8 @@ var
 
 module.exports = {
 	rules: {
-		'newline-around-multiline-blocks': function(context) {
-			return _.object(_.map([
-				'ArrayExpression',
-				'BlockStatement',
-				'ObjectExpression'
-			], function(R) { return [R, function(node) {
-				if(node.loc.start.line === node.loc.end.line)
-					return;
-
-				_.each(node.body || node.elements || node.properties, function(N, I, L) {
-					var
-						isMultiline = Boolean(N.loc.end.line - N.loc.start.line),
-						next = L[I + 1],
-						nextCommentsLength = 0,
-						nextIsMultiline;
-
-					if(!next)
-						return;
-
-					nextIsMultiline = Boolean(next.loc.end.line - next.loc.start.line);
-
-					// Keep in mind that next node may lead by multiple comments nodes
-					if(next.leadingComments)
-						nextCommentsLength = _.last(next.leadingComments).loc.end.line - _.first(next.leadingComments).loc.start.line + 1;
-
-					if(isMultiline && (next.loc.start.line - N.loc.end.line) !== (2 + nextCommentsLength))
-						context.report(N, {line: N.loc.end.line}, 'Multiline node should have trailing new line');
-
-					if(isMultiline)
-						return;
-
-					if(nextIsMultiline && next.loc.start.line !== N.loc.end.line && (next.loc.start.line - N.loc.end.line) != (2 + nextCommentsLength))
-						context.report(N, {line: N.loc.end.line}, 'Singleline node prior to multiline should have trailing new line');
-
-					if(!N.leadingComments && !nextIsMultiline && !nextCommentsLength && (next.loc.start.line - N.loc.end.line) > 1)
-						context.report(N, {line: N.loc.end.line + 1}, 'There should not be new line between singleline nodes');
-				});
-			}]; }));
-		},
-
-		'no-var-on-same-line': function(context) {
-			return {
-				'Program': function(node) {
-					var
-						src = context.getSource(),
-						re = new RegExp('var[^\n]+$', 'mg'),
-						match,
-						lines;
-					
-					while((match = re.exec(src)) !== null) {
-						lines = src.slice(0, re.lastIndex).split(/\r?\n/g);
-						
-						context.report(node, {
-							line: lines.length,
-							column: lines[lines.length - 1].length - match[0].length + 1
-						}, 'var statement required to take the whole line');
-					}
-				}
-			}
-		},
-		
-		'no-full-name-lambda-params': function(context) {
-			return {
-				'FunctionExpression': function(node) {
-					var
-						lambdaParamRE = new RegExp('^[A-Z]{1,2}$')
-					
-					// Allow full name params in object properties
-					if(node.parent.type !== 'CallExpression' || node.id)
-						return;
-					
-					if(node.params.map(function(P) {
-						return P.type === 'Identifier' && lambdaParamRE.exec(P.name) === null && P.name;
-					}).filter(Boolean).length)
-						context.report(node,'Only 1-2 capital letters names allowed for lambda params');
-				}
-			}
-		}
+		'newline-around-multiline-blocks': require("./lib/rules/newline-around-multiline-blocks"),
+		'no-var-on-same-line': require("./lib/rules/no-var-on-same-line"),
+		'no-full-name-lambda-params': require("./lib/rules/no-full-name-lambda-params")
 	}
 };

--- a/index.js
+++ b/index.js
@@ -2,8 +2,8 @@
 
 module.exports = {
 	rules: {
-		'newline-around-multiline-blocks': require("./lib/rules/newline-around-multiline-blocks"),
-		'no-var-on-same-line': require("./lib/rules/no-var-on-same-line"),
-		'no-full-name-lambda-params': require("./lib/rules/no-full-name-lambda-params")
+		'newline-around-multiline-blocks': require('./lib/rules/newline-around-multiline-blocks'),
+		'no-var-on-same-line': require('./lib/rules/no-var-on-same-line'),
+		'no-full-name-lambda-params': require('./lib/rules/no-full-name-lambda-params)
 	}
 };

--- a/index.js
+++ b/index.js
@@ -1,8 +1,5 @@
 'use strict';
 
-var
-	_ = require('underscore');
-
 module.exports = {
 	rules: {
 		'newline-around-multiline-blocks': require("./lib/rules/newline-around-multiline-blocks"),

--- a/lib/rules/newline-around-multiline-blocks.js
+++ b/lib/rules/newline-around-multiline-blocks.js
@@ -1,16 +1,7 @@
-/**
- * @fileoverview
- * @author
- */
-
 "use strict";
 
 var
 	_ = require('underscore');
-
-//------------------------------------------------------------------------------
-// Rule Definition
-//------------------------------------------------------------------------------
 
 module.exports = function(context) {
 	return _.object(_.map([

--- a/lib/rules/newline-around-multiline-blocks.js
+++ b/lib/rules/newline-around-multiline-blocks.js
@@ -51,5 +51,3 @@ module.exports = function(context) {
 		});
 	}]; }));
 };
-
-module.exports.schema = [];

--- a/lib/rules/newline-around-multiline-blocks.js
+++ b/lib/rules/newline-around-multiline-blocks.js
@@ -1,0 +1,55 @@
+/**
+ * @fileoverview
+ * @author
+ */
+
+"use strict";
+
+var
+	_ = require('underscore');
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+	return _.object(_.map([
+		'ArrayExpression',
+		'BlockStatement',
+		'ObjectExpression'
+	], function(R) { return [R, function(node) {
+		if(node.loc.start.line === node.loc.end.line)
+			return;
+
+		_.each(node.body || node.elements || node.properties, function(N, I, L) {
+			var
+				isMultiline = Boolean(N.loc.end.line - N.loc.start.line),
+				next = L[I + 1],
+				nextCommentsLength = 0,
+				nextIsMultiline;
+
+			if(!next)
+				return;
+
+			nextIsMultiline = Boolean(next.loc.end.line - next.loc.start.line);
+
+			// Keep in mind that next node may lead by multiple comments nodes
+			if(next.leadingComments)
+				nextCommentsLength = _.last(next.leadingComments).loc.end.line - _.first(next.leadingComments).loc.start.line + 1;
+
+			if(isMultiline && (next.loc.start.line - N.loc.end.line) !== (2 + nextCommentsLength))
+				context.report(N, {line: N.loc.end.line}, 'Multiline node should have trailing new line');
+
+			if(isMultiline)
+				return;
+
+			if(nextIsMultiline && next.loc.start.line !== N.loc.end.line && (next.loc.start.line - N.loc.end.line) != (2 + nextCommentsLength))
+				context.report(N, {line: N.loc.end.line}, 'Singleline node prior to multiline should have trailing new line');
+
+			if(!N.leadingComments && !nextIsMultiline && !nextCommentsLength && (next.loc.start.line - N.loc.end.line) > 1)
+				context.report(N, {line: N.loc.end.line + 1}, 'There should not be new line between singleline nodes');
+		});
+	}]; }));
+};
+
+module.exports.schema = [];

--- a/lib/rules/no-full-name-lambda-params.js
+++ b/lib/rules/no-full-name-lambda-params.js
@@ -1,16 +1,7 @@
-/**
- * @fileoverview
- * @author
- */
-
 "use strict";
 
 var
 	_ = require('underscore');
-
-//------------------------------------------------------------------------------
-// Rule Definition
-//------------------------------------------------------------------------------
 
 module.exports = function(context) {
 	return {

--- a/lib/rules/no-full-name-lambda-params.js
+++ b/lib/rules/no-full-name-lambda-params.js
@@ -1,0 +1,33 @@
+/**
+ * @fileoverview
+ * @author
+ */
+
+"use strict";
+
+var
+	_ = require('underscore');
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+	return {
+		'FunctionExpression': function(node) {
+			var
+				lambdaParamRE = new RegExp('^[A-Z]{1,2}$')
+
+			// Allow full name params in object properties
+			if(node.parent.type !== 'CallExpression' || node.id)
+				return;
+
+			if(node.params.map(function(P) {
+				return P.type === 'Identifier' && lambdaParamRE.exec(P.name) === null && P.name;
+			}).filter(Boolean).length)
+				context.report(node,'Only 1-2 capital letters names allowed for lambda params');
+		}
+	}
+};
+
+module.exports.schema = [];

--- a/lib/rules/no-full-name-lambda-params.js
+++ b/lib/rules/no-full-name-lambda-params.js
@@ -29,5 +29,3 @@ module.exports = function(context) {
 		}
 	}
 };
-
-module.exports.schema = [];

--- a/lib/rules/no-var-on-same-line.js
+++ b/lib/rules/no-var-on-same-line.js
@@ -1,16 +1,7 @@
-/**
- * @fileoverview
- * @author
- */
-
 "use strict";
 
 var
 	_ = require('underscore');
-
-//------------------------------------------------------------------------------
-// Rule Definition
-//------------------------------------------------------------------------------
 
 module.exports = function(context) {
 	return {

--- a/lib/rules/no-var-on-same-line.js
+++ b/lib/rules/no-var-on-same-line.js
@@ -32,5 +32,3 @@ module.exports = function(context) {
 		}
 	}
 };
-
-module.exports.schema = [];

--- a/lib/rules/no-var-on-same-line.js
+++ b/lib/rules/no-var-on-same-line.js
@@ -1,0 +1,36 @@
+/**
+ * @fileoverview
+ * @author
+ */
+
+"use strict";
+
+var
+	_ = require('underscore');
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+	return {
+		'Program': function(node) {
+			var
+				src = context.getSource(),
+				re = new RegExp('var[^\n]+$', 'mg'),
+				match,
+				lines;
+
+			while((match = re.exec(src)) !== null) {
+				lines = src.slice(0, re.lastIndex).split(/\r?\n/g);
+
+				context.report(node, {
+					line: lines.length,
+					column: lines[lines.length - 1].length - match[0].length + 1
+				}, 'var statement required to take the whole line');
+			}
+		}
+	}
+};
+
+module.exports.schema = [];

